core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBundleLoader.java

```diff
@@ -64,14 +62,26 @@ public CatalogBundleLoader(Predicate<Bundle> applicationsPermitted, ManagementCo
         this.managementContext = managementContext;
     }
 
+    public void scanForCatalog(Bundle bundle, boolean force, boolean validate) {
+        scanForCatalogInternal(bundle, force, validate, false);
+    }
+    
     /**
      * Scan the given bundle for a catalog.bom and adds it to the catalog.
      *
      * @param bundle The bundle to add
      * @return A list of items added to the catalog
      * @throws RuntimeException if the catalog items failed to be added to the catalog
      */
-    public Iterable<? extends CatalogItem<?, ?>> scanForCatalog(Bundle bundle) {
+    public Iterable<? extends CatalogItem<?, ?>> scanForCatalogLegacy(Bundle bundle) {
```

Comment:
Now that you've removed `CatalogBomScanner` this method isn't used any more, I think we can get rid of it and the one below.  See also comments on `OsgiManager.java`.  I think we can also get rid of `applicationsPermitted` in this class, in the one place that its constructor is called the predicate is hardcoded to `true`. The interesting case is on line 98 below.